src/deploy/mod.rs

```diff
@@ -1,75 +1,38 @@
-mod route;
-use route::publish_routes;
+mod schedule;
+mod zoned;
+mod zoneless;
 
-use crate::commands::subdomain::Subdomain;
-use crate::http;
-use crate::settings::global_user::GlobalUser;
-use crate::settings::toml::{DeployConfig, Zoneless};
-
-pub fn worker(
-    user: &GlobalUser,
-    deploy_config: &DeployConfig,
-) -> Result<Vec<String>, failure::Error> {
-    match deploy_config {
-        DeployConfig::Zoneless(zoneless_config) => {
-            // this is a zoneless deploy
-            log::info!("publishing to workers.dev subdomain");
-            let deploy_address = publish_zoneless(user, zoneless_config)?;
-            let addresses = vec![deploy_address];
-            Ok(addresses)
-        }
-        DeployConfig::Zoned(zoned_config) => {
-            // this is a zoned deploy
-            log::info!("publishing to zone {}", zoned_config.zone_id);
+pub use schedule::ScheduleTarget;
+pub use zoned::ZonedTarget;
+pub use zoneless::ZonelessTarget;
 
-            let published_routes = publish_routes(&user, zoned_config)?;
+use crate::settings::global_user::GlobalUser;
 
-            let addresses: Vec<String> =
-                published_routes.iter().map(|r| format!("{}", r)).collect();
+/// A set of deploy targets.
+pub type DeploymentSet = Vec<DeployTarget>;
 
-            Ok(addresses)
-        }
-    }
+#[derive(Debug, PartialEq, Clone)]
+pub enum DeployTarget {
```

Comment:
@ashleymichal So it turns out removing all type information by using a boxed trait object makes testing and checks in `wrangler dev` really hard so I decided to just use an enum instead to simplify a lot of stuff...